w3lib/encoding.py

```diff
@@ -173,15 +174,15 @@ def read_bom(data):
 
 # Python decoder doesn't follow unicode standard when handling
 # bad utf-8 encoded strings. see http://bugs.python.org/issue8271
-codecs.register_error('w3lib_replace', lambda exc: (u'\ufffd', exc.start+1))
+codecs.register_error('w3lib_replace', lambda exc: (u'\ufffd', exc.end))
 
 def to_unicode(data_str, encoding):
     """Convert a str object to unicode using the encoding given
 
     Characters that cannot be converted will be converted to ``\\ufffd`` (the
     unicode replacement character).
     """
-    return data_str.decode(encoding, 'w3lib_replace')
+    return data_str.decode(encoding, 'replace' if six.PY34 else 'w3lib_replace')
```

Comment:
why py3.4 is an exception here?

Reply:
It is a badly named (and undocumented) variable from six which means >= 3.4 (see https://github.com/benjaminp/six/issues/147).

Reply:
+1 to use a more readable check

Reply:
I am fine as long as we left a comment and link to that issue on what PY34 means. May be in future versions we can replace it by what six decides to use.

Reply:
The 'replace' error handling was fixed in 3.3, so that's the reason I added this check. Now I'm checking version_info directly.